src/newrelic/alerts.js

```diff
@@ -151,15 +180,20 @@ async function getPolicies(auth, policyName) {
       json: true,
     });
 
-    const policies = response.policies.map(({ id, name }) => ({ id, name }));
+    let policy;
+    const allPolicies = response.policies
+      ? response.policies.map(({ id, name }) => ({ id, name }))
+      : [];
     if (policyName) {
-      return policies.filter((policy) => policy.name === policyName);
-    } else {
-      return [];
+      [policy] = allPolicies.filter((pol) => pol.name === policyName);
```

Comment:
```suggestion
      policy = allPolicies.find((pol) => pol.name === policyName);
```

I'd also try to replace the `let` above with a `const`…